Rules.py

```diff
@@ -910,3 +1435,76 @@ def get_rule_to_add(region):
                 continue
 
             add_rule(location, get_rule_to_add(location.parent_region))
+
+def set_inverted_bunny_rules(world, player):
+
+    # regions for the exits of multi-entrace caves/drops that bunny cannot pass
+    # Note spiral cave may be technically passible, but it would be too absurd to require since OHKO mode is a thing.
+    bunny_impassable_caves = ['Bumper Cave', 'Two Brothers House', 'Hookshot Cave', 'Skull Woods First Section (Right)', 'Skull Woods First Section (Left)', 'Skull Woods First Section (Top)', 'Turtle Rock (Entrance)', 'Turtle Rock (Second Section)', 'Turtle Rock (Big Chest)', 'Skull Woods Second Section (Drop)',
+                              'Turtle Rock (Eye Bridge)', 'Sewers', 'Pyramid', 'Spiral Cave (Top)', 'Desert Palace Main (Inner)', 'Fairy Ascension Cave (Drop)', 'The Sky']
+
+    bunny_accessible_locations = ['Link\'s Uncle', 'Sahasrahla', 'Sick Kid', 'Lost Woods Hideout', 'Lumberjack Tree', 'Checkerboard Cave', 'Potion Shop', 'Spectacle Rock Cave', 'Pyramid', 'Hype Cave - Generous Guy', 'Peg Cave', 'Bumper Cave Ledge', 'Dark Blacksmith Ruins', 'Bombos Tablet', 'Ether Tablet', 'Purple Chest']
+
+
+    def path_to_access_rule(path, entrance):
+        return lambda state: state.can_reach(entrance) and all(rule(state) for rule in path)
+
+    def options_to_access_rule(options):
+        return lambda state: any(rule(state) for rule in options)
+
+    def get_rule_to_add(region):
+        if not region.is_dark_world:
+            return lambda state: state.has_Pearl(player)
+        # in this case we are mixed region.
+        # we collect possible options.
+
+        # The base option is having the moon pearl
+        possible_options = [lambda state: state.has_Pearl(player)]
+
+        # We will search entrances recursively until we find
+        # one that leads to an exclusively light world region
+        # for each such entrance a new option is added that consist of:
+        #    a) being able to reach it, and
+        #    b) being able to access all entrances from there to `region`
```

Comment:
Probably worth updating these comments to switch light and dark work vs the non-inverted version of the function. Otherwise they can be extremely confusing.

Reply:
Updated